File: Task3 (ATM Interface)/frontend/src/service/atmService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/atm';

export const withdraw = async (accountNumber, amount) => {
  try {
    const response = await axios.post(`${API_URL}/withdraw`, null, {
      params: { accountNumber, amount }
    });
    return response.data;
  } catch (error) {
    console.error('Withdrawal error:', error);
    throw error;
  }
};

export const deposit = async (accountNumber, amount) => {
  try {
    const response = await axios.post(`${API_URL}/deposit`, null, {
      params: { accountNumber, amount }
    });
    return response.data;
  } catch (error) {
    console.error('Deposit error:', error);
    throw error;
  }
};

export const checkBalance = async (accountNumber) => {
  try {
    const response = await axios.get(`${API_URL}/balance`, {
      params: { accountNumber }
    });
    return response.data;
  } catch (error) {
    console.error('Balance check error:', error);
    throw error;
  }
};

export const validatePin = async (accountNumber, pin) => {
  try {
    const response = await axios.post(`${API_URL}/validate`, null, {
      params: { accountNumber, pin }
    });
    return response.data;
  } catch (error) {
    console.error('PIN validation error:', error);
    throw error;
  }
};

File: Task3 (ATM Interface)/backend/src/main/java/com/task3/atmInterface/Service/AtmService.java
package com.task3.atmInterface.Service;

import com.task3.atmInterface.Model.Account;
import com.task3.atmInterface.Model.AtmResponse;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AtmService {

    @Autowired
    private AccountService accountService;

    public AtmResponse withdraw(String accountNumber, double amount) {
        Account account = accountService.getAccount(accountNumber);

        if(account == null){
            return new AtmResponse(false, "Account Not Found", 0);
        }
        if(account.getBalance() < amount){
            return new AtmResponse(false, "Insufficient Balance", account.getBalance());
        }
        account.setBalance(account.getBalance() - amount);
        return new AtmResponse(true, "Withdrawal Successful", account.getBalance());
    }

    public AtmResponse deposit(String accountNumber, double amount) {
        Account account = accountService.getAccount(accountNumber);

        if(account == null){
            return new AtmResponse(false, "Account Not Found", 0);
        }
        account.setBalance(account.getBalance() + amount);
        return new AtmResponse(true, "Deposit Successful", account.getBalance());
    }

    public AtmResponse checkBalance(String accountNumber) {
        Account account = accountService.getAccount(accountNumber);
        if (account == null) {
            return new AtmResponse(false, "Account not found", 0);
        }
        return new AtmResponse(true, "Balance checked", account.getBalance());
    }

    public boolean validatePin(String accountNumber, String pin) {
        Account account = accountService.getAccount(accountNumber);
        return account != null && account.validatePin(pin);
    }
}
